megamek/src/megamek/client/bot/princess/Princess.java

```diff
@@ -79,10 +81,15 @@
     private final IHonorUtil honorUtil = new HonorUtil();
 
     private boolean initialized = false;
-
-    //private PathSearcher pathSearcher;
-    private BasicPathRanker pathRanker;
-    private FireControl fireControl;
+   
+    // path rankers and fire controls, organized by their explicitly given types to avoid confusion
+    private HashMap<PathRankerType, IPathRanker> pathRankers;
```

Comment:
Most changes in Princess are the introduction of a variety of fire control and path ranker class instances.